zkevm-circuits/src/util/cell_manager.rs

```diff
@@ -1,8 +1,119 @@
-trait CellManagerStrategy {
-    fn query_top_aligned_cell(&mut self, cell_type: CellType) -> Cell<F>;
-    fn query_cell(&mut self, cell_type: CellType) -> Cell<F>;
-    fn query_cells(&mut self, cell_type: CellType, count: usize) -> Vec<Cell<F>>;
-    fn columns(&self) -> &[CellColumn<F>];
-    fn get_width(&self) -> usize;
+use std::{collections::HashMap, marker::PhantomData};
+
+use halo2_proofs::plonk::{Advice, Column};
+
+use crate::evm_circuit::table::Table;
+
+#[derive(Eq, Hash, PartialEq)]
+/// CellType
+pub(crate) enum CellType {
+    StoragePhase1,
+    StoragePhase2,
+    StoragePermutation,
+    LookupByte,
+    Lookup(Table),
+}
+
+/// Cell
+pub struct Cell<F> {
+    _p: PhantomData<F>,
+}
+
+/// CellColumn
+pub struct CellColumn<F> {
+    _p: PhantomData<F>,
+}
+
+/// CellManagerStrategy
+pub(crate) trait CellManagerStrategy<F> {
+    // /// query_top_aligned_cell
+    // fn query_top_aligned_cell(&mut self, cell_type: CellType) -> Cell<F>;
+    // query_cell
+    // fn query_cell(&mut self, cell_type: CellType) -> Cell<F>;
+    // query_cells
+    // fn query_cells(&mut self, cell_type: CellType, count: usize) -> Vec<Cell<F>>;
+    // columns
+    // fn columns(&self) -> &[CellColumn<F>];
+    // get_width
+    // fn get_width(&self) -> usize;
+    // get_height
+    // fn get_height(&self) -> usize;
+
+    /// on_creation
+    fn on_creation(&mut self, columns: &mut CellManagerColumns);
+    /// query_cell
+    fn query_cell(&mut self, columns: &mut CellManagerColumns, cell_type: CellType);
+    /// get_height
     fn get_height(&self) -> usize;
 }
+
+#[derive(Default)]
+pub(crate) struct CellManagerColumns {
+    columns: HashMap<CellType, Vec<Column<Advice>>>,
+}
+
+impl CellManagerColumns {
+    pub fn add_column(&mut self, cell_type: CellType, column: Column<Advice>) {
```

Comment:
Because with the cell manager a column is used by different signals, this would not be useful. We need to create a MockProver that is Cell or signal aware to put annotations that would make sense. We talk about this with @ed255